app/knights/knight.py

```diff
@@ -0,0 +1,25 @@
+class Knight:
+
+    def __init__(self, config: dict) -> None:
+
+        self.name = config.get("name")
+        self.power = config.get("power")
+        self.hp = config.get("hp")
+        self.protection = 0
+
+        if config["armour"] and len(config["armour"]) > 0:
+            for item in config["armour"]:
+                self.protection += item["protection"]
+
+        if config["weapon"]["power"]:

```

Comment:
The code assumes that 'weapon' is always present in the config dictionary and that it contains a 'power' key. Consider using `config.get('weapon', {}).get('power', 0)` to avoid a `KeyError` if 'weapon' or 'power' is missing.